include/ginkgo/core/base/combination.hpp

```diff
@@ -112,6 +112,7 @@ class Combination : public EnableLinOp<Combination<ValueType>>,
         add_operators(std::forward<Rest>(rest)...);
     }
 
+protected:
```

Comment:
I agree with @upsj that we should not expose this member. I think it would be enough to add a constructor, which takes the iterators, and just an dimension as an extra parameter. Then you could build the linop from that parameter, and just check that the operator dimensions match up.